CONTRIBUTING.md

```diff
@@ -0,0 +1,116 @@
+# Contributing
+
+Implementing and maintaining a protocol stack is a lot of work, therefore any
+help is appreciated, from creating issues, to contributing documentation, 
+fixing issues and adding new features.
+
+Please follow the best-practice contribution guidelines as mentioned below 
+when submitting any changes.
+
+### Contributions License Agreement
+
+Before contributing, please read the 
+[Contributions License Agreement](./CLA.md).
+Every contributor must sign this agreement before any code is merged into the
+repository.
+
+### Git
+
+Please send us the pull request including details about your change and 
+referencing relevant issues or discussions.
+
+In the case that write access to the main repository were granted to 
+contributors the following basic ground-rules should be applied:
+
+* Do not use the main branch for ongoing work. Please use feature- or 
+fix-branches instead
+
+### Guidelines for contributions
+
+The following general guidelines for contributions apply:
+* If you plan to introduce new features, changes to the somwhoe "magic" typings
+or significant changes, the discussions in GitHub are a good idea to discuss 
+details together with the community before starting to work on it. This can ease
+the PR review provess because all parties know more context.
+* The library tries to have "no external dependencies" to be used in also other 
+JavaScript environments than just Node.js. Please do not introduce new 
+dependencies without a discussion before committing the PR.
+* If you had a discussion on the topic with one developer please make sure to 
+include this developer in any case in your review list. When the PR got 
+approved it will be merged by the reviewer.
+
+In the case that write access to the main repository were granted to 
+contributors the following rules should be applied:
+
+* External API changes and significant modifications ought to be subject to 
+an internal pull-request to solicit feedback from other contributors.
+* Internal pull-requests to solicit feedback are encouraged for any other 
+non-trivial contribution but left to the discretion of the contributor.
+* As a reviewer, upon approval, don't merge the PR, let the author do the 
+final merge and associate post merge tasks if any.
+
+"Internal-pull-request" means that also with direct commit access to the 
```

Comment:
I still find the notion of "internal pull request" confusing...
I would leave that out and we can add more guidance on this if we have issues...